lnd_services.go

```diff
@@ -766,3 +767,58 @@ func getClientConn(cfg *LndServicesConfig) (*grpc.ClientConn, error) {
 
 	return conn, nil
 }
+
+// GetTLSCredentials gets the tls credentials, whether provided as straight-up
+// data or a path to a certificate file.
+func GetTLSCredentials(tlsData, tlsPath string) (
+	credentials.TransportCredentials, error) {
+
+	if tlsPath != "" && tlsData != "" {
+		return nil, fmt.Errorf("must set only one: TLSPath or TLSData")
+	}
+
+	var creds credentials.TransportCredentials
+	var err error
+
+	// We'll determine if the tls certificate is passed in directly as
+	// data, by a path, or try the system's certificate chain, and then
+	// load it.
+	switch {
+	case tlsData != "":
+		tlsBytes := []byte(tlsData)
+
+		block, _ := pem.Decode(tlsBytes)
+		if block == nil || block.Type != "CERTIFICATE" {
+			return nil, errors.New("failed to decode PEM block " +
+				"containing tls certificate")
+		}
+
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return nil, err
+		}
+
+		pool := x509.NewCertPool()
+		pool.AddCert(cert)
+
+		// Load the specified TLS certificate and build transport
+		// credentials.
+		creds = credentials.NewClientTLSFromCert(pool, "")
+	case tlsPath != "":
+		if tlsPath == "" {
+			tlsPath = defaultTLSCertPath
+		}
```

Comment:
Impossible to hit this since we've selected on `tlsPath != ""`

Reply:
Ah good catch. I assume we still want to support the case where the tlsPath isn't set, but the user is relying on whatever the defaultTLSCertPath is set to. So I added that logic to the default case, if you want to check that it looks okay